cms/server/ContestWebServer.py

```diff
@@ -1061,10 +1063,12 @@ def which_language(user_filename):
                              if it is not a recognized language.
 
             """
-            for source_ext, language in SOURCE_EXT_TO_LANGUAGE_MAP.iteritems():
-                if user_filename.endswith(source_ext):
-                    return language
-            return None
+
+            languageSubmitted = self.get_argument("submission_lang", "")
```

Comment:
How does a user not set "submission_lang", doesn't that mean they are messing around with the html? If that's the case, is it okay to get Tornado to give them a 400 Error instead?